plugins/org.eclipse.n4js.utils.ui/src/org/eclipse/n4js/utils/ui/editor/AvoidRefreshDocumentProvider.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.utils.ui.editor;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.xtext.ui.editor.model.XtextDocumentProvider;
+
+/**
+ * This class temporarily solves the problem that a file is refreshed when opened.
+ */
+public class AvoidRefreshDocumentProvider extends XtextDocumentProvider {
+
+	@Override
+	protected void refreshFile(IFile file, IProgressMonitor monitor) throws CoreException {
```

Comment:
Please add a comment: file.isSynchronized does not require a scheduling rule and thus helps to identify a no-op attempt to refresh the file. The no-op will otherwise be blocked by a running build or cancel a running build